students/adam_strong/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+'''
+Returns total price paid for individual rentals:
+   argparse options:
+
+        required: -i INPUT FILE -o OUTPUT FILE
+        optional: -d # (Debug levels:
+                        0 - No error logging (DEFAULT)
+                        1 - Only error messages (causing it to crash)
+                        2 - Only warnings (skipped records, how many errors)
+                        3 - All warnings (including script flow indicators)
+'''
+import argparse
+import json
+import datetime
+import math
+import sys
+import logging
+
+logger = logging.getLogger()
+LOG_FORMAT = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+log_file = 'logs/charges_calc' + datetime.datetime.now().strftime('%Y-%m-%d')+'.log'
+formatter = logging.Formatter(LOG_FORMAT)
+logger.setLevel(logging.DEBUG)
+
+
+# MAIN CODE___________________________________________________________________
+
+def parse_cmd_arguments():
+    '''Interpret the command line arguments, -i INPUT, -o OUTPUT, -d DEBUG'''
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-debug', '-d', type=int, default=0, required=False,
+                        help='Debug Level: 0=No debug messages or log file, \
+                        1=Only error messages, 2=Error messages and warnings,\
+                        3=Error messages, warnings and debug messages')
+    try:
+        return parser.parse_args()
+    except TypeError:
+        sys.exit()
+
+def set_logger(debug):
+    '''Sets the logging level input from command line'''
+    if debug > 0:
+        if debug == 1:
```

Comment:
There is one thing missing here: As coded, this will also accept any invalid value (such as -d 10). Try to add a condition to error out if the user enters an invalid debug level.